embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/network/logging/EmbraceNetworkLoggingService.kt

```diff
@@ -49,113 +45,40 @@ internal class EmbraceNetworkLoggingService(
 
     private var domainSuffixCallLimits = configService.networkBehavior.getNetworkCallLimitsPerDomainSuffix()
 
-    fun logNetworkCall(
-        callId: String,
-        url: String,
-        httpMethod: String,
-        statusCode: Int,
-        startTime: Long,
-        endTime: Long,
-        bytesSent: Long,
-        bytesReceived: Long,
-        traceId: String?,
-        w3cTraceparent: String?,
-        networkCaptureData: NetworkCaptureData?
-    ) {
-        val duration = max(endTime - startTime, 0)
-        val strippedUrl = stripUrl(url)
-        val validTraceId = getValidTraceId(traceId)
-        val networkCall = NetworkCallV2(
-            url = strippedUrl,
-            httpMethod = httpMethod,
-            responseCode = statusCode,
-            bytesSent = bytesSent,
-            bytesReceived = bytesReceived,
-            startTime = startTime,
-            endTime = endTime,
-            duration = duration,
-            traceId = validTraceId,
-            w3cTraceparent = w3cTraceparent
-        )
-
-        if (networkCaptureData != null) {
-            networkCaptureService.logNetworkCapturedData(
-                url,
-                httpMethod,
-                statusCode,
-                startTime,
-                endTime,
-                networkCaptureData
-            )
-        }
-
-        processNetworkCall(callId, networkCall)
+    override fun logNetworkRequest(networkRequest: EmbraceNetworkRequest) {
+        logNetworkCaptureData(networkRequest)
+        processNetworkRequest(networkRequest)
     }
 
-    fun logNetworkError(
-        callId: String,
-        url: String,
-        httpMethod: String,
-        startTime: Long,
-        endTime: Long,
-        errorType: String?,
-        errorMessage: String?,
-        traceId: String?,
-        w3cTraceparent: String?,
-        networkCaptureData: NetworkCaptureData?
-    ) {
-        val duration = max(endTime - startTime, 0)
-        val strippedUrl = stripUrl(url)
-        val validTraceId = getValidTraceId(traceId)
-        val networkCall = NetworkCallV2(
-            url = strippedUrl,
-            httpMethod = httpMethod,
-            startTime = startTime,
-            endTime = endTime,
-            duration = duration,
-            traceId = validTraceId,
-            w3cTraceparent = w3cTraceparent,
-            errorMessage = errorMessage,
-            errorType = errorType
-        )
-
-        if (networkCaptureData != null) {
+    private fun logNetworkCaptureData(networkRequest: EmbraceNetworkRequest) {
+        if (networkRequest.networkCaptureData != null) {
             networkCaptureService.logNetworkCapturedData(
-                url,
-                httpMethod,
-                NETWORK_ERROR_CODE,
-                startTime,
-                endTime,
-                networkCaptureData,
-                errorMessage
+                networkRequest.url, //TODO: This used the non-stripped URL, is that correct?
```

Comment:
Yeah, we want to log the entire URL when capturing the entire request/response in case there are parameters are that important. We aren't aggregating by this so we don't need to reduce the cardinality by tweaking the URL